addons/mass_mailing/static/src/js/mass_mailing_full_width_form.js

```diff
@@ -1,10 +1,150 @@
 /** @odoo-module **/
 
 import FormView from 'web.FormView';
+import FormController from 'web.FormController';
 import FormRenderer from 'web.FormRenderer';
+import { bus, _t } from 'web.core';
 import viewRegistry from 'web.view_registry';
 import config from 'web.config';
 
+const MassMailingFullWidthFormController = FormController.extend({
+    custom_events: _.extend({}, FormController.prototype.custom_events,{
+        iframe_updated: '_onIframeUpdated',
+    }),
+
+    /**
+     * @override
+     */
+    init() {
+        this._super(...arguments);
+        this._boundOnDomUpdated = this._onDomUpdated.bind(this);
+        bus.on('DOM_updated', this, this._onDomUpdated);
+        this._resizeObserver = new ResizeObserver(this._onResizeIframeContents.bind(this));
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        bus.off('DOM_updated', this, this._boundOnDomUpdated);
+        this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Resize the mailing editor's iframe container so its height fits its
+     * contents. This needs to be called whenever the iframe's contents might
+     * have changed, eg. when adding/removing content to/from it or when a
+     * template is picked.
+     *
+     * @private
+     * @param {JQuery} $iframe
+     */
+    _resizeMailingEditorIframe() {
+        const VERTICAL_OFFSET = 12; // Vertical offset picked for visual design purposes.
+        const minHeight = $(window).height() - Math.abs(this.$iframe.offset().top) - (VERTICAL_OFFSET / 2);
+        const $iframeDoc = this.$iframe.contents();
+        const $themeSelectorNew = $iframeDoc.find('.o_mail_theme_selector_new');
+        if ($themeSelectorNew.length) {
+            this.$iframe.height(Math.max($themeSelectorNew[0].scrollHeight + VERTICAL_OFFSET, minHeight));
+        } else {
+            const ref = $iframeDoc.find('#iframe_target')[0];
+            if (ref) {
+                this.$iframe.height(Math.max(ref.scrollHeight + VERTICAL_OFFSET, minHeight));
+            }
+        }
+    },
+    /**
+     * Return true if the mailing editor is in full screen mode, false
+     * otherwise.
+     *
+     * @private
+     * @returns {boolean}
+     */
+   _isFullScreen() {
```

Comment:
Linting: indentation issue (seems fixed in next commit ? probably a rebase to do to avoid commit fixups)